WcaOnRails/spec/controllers/results_submission_controller_spec.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ResultsSubmissionController, type: :controller do
+  let(:results_file) { fixture_file_upload '/files/results.json', 'application/json' }
+  let(:submission_message) { "Hello, here are the results" }
+
+  before :all do
+    delegate = FactoryBot.create :delegate
+    @comp = FactoryBot.create(:competition, id: @competition_id, delegates: [delegate])
+  end
```

Comment:
What is this `@competition_id` variable here? Is it actually equal to something, or is it `nil`?

I believe this can be rewritten as:

```
let(:delegate) { FactoryBot.create :delegate }
let(:comp) { FactoryBot.create(:competition, delegates: [delegate]) }
```

(some of your later references to `@comp` will have to change to just `comp`).

Also, as @jonatanklosko said, this should be a request spec, maybe focus on that before working on my comments.